include/eepromfs.h

```diff
@@ -0,0 +1,139 @@
+ * @see #eepfs_config_t
+ * @see #eepfs_init
+ */
+typedef struct eepfs_entry
+{
+    /**
+     * @brief File path.
+     * 
+     * This cannot be NULL and must be a `\0`-terminated string.
+     * 
+     * There are no enforced limitations on directory structure
+     * or file naming conventions except that all paths within
+     * the filesystem must be unique and at least one character.
+     * 
+     * A leading '/' is optional and will be ignored if set.
+     * Files accessed through stdio will always have the leading
+     * '/', but files accessed through #eepfs_open can omit it.
+     * 
+     * The filesytem does not support entries for directories,
+     * nor does it support listing files in a given directory.
+     */
+    const char * const path;
+    /**
+     * @brief File size in bytes.
+     * 
+     * In order to make the most use of limited EEPROM space,
+     * files should be (but are not required to be) aligned to the
+     * 8-byte block size. Unaligned bytes at the end of a file will
+     * be wasted as padding; files must start on a block boundary.
+     * 
+     * The filesystem itself reserves the first block of EEPROM,
+     * so your total filesystem size cannot exceed the available
+     * EEPROM size minus 8 bytes (64 bits):
+     * 
+     * * 4k EEPROM: 512 - 8 = 504 bytes (63 blocks) free.
+     * * 16k EEPROM: 2048 - 8 = 2040 bytes (255 blocks) free.
+     */
+    const size_t size;
+} eepfs_entry_t;
+
+/**
+ * @brief EEPROM filesystem configuration options
+ * @see #eepfs_init
+ */
+typedef struct eepfs_config
+{
+    /** @brief Array of file entries; see #eepfs_entry_t */
+    const eepfs_entry_t * const files;
+    /**
+     * @brief Number of file entries in the array
+     * 
+     * Each file will take up a minimum of 1 block, plus the
+     * filesystem itself reserves the first block of EEPROM,
+     * so the number of files in your filesystem has a practical
+     * limit of the number of available EEPROM blocks minus 1:
+     * 
+     * * 4k EEPROM: 63 files maximum.
+     * * 16k EEPROM: 255 files maximum.
+     * 
+     * In practice, this limitation is unlikely to matter since
+     * most files will span multiple blocks and run up against
+     * the storage capacity limits of EEPROM first.
+     */
+    const size_t num_files;
+    /**
+     * @brief POSIX standard input and output filesystem prefix.
+     * 
+     * This can be NULL if you only want to use `eepfs_*` functions,
+     * but setting this (e.g. "eeprom:/") allows POSIX file operations
+     * to access the EEPROM filesystem with stdio library functions.
+     */
+    const char * const stdio_prefix;
+    /**
+     * @brief Whether to validate EEPROM data during initialization.
+     * 
+     * If this is false and the signature does not match,
+     * all EEPROM data will be erased!
+     * 
+     * @see #eepfs_check_signature
+     */
+    const bool skip_signature_check;
+} eepfs_config_t;
```

Comment:
### Regarding `const` everywhere

No objection to removing `const` from the public struct declarations. My heuristic is to `const` everything unless there's a good reason not to, but you're right that it's unnecessary and potentially annoying here.

### Regarding `stdio_prefix`

No objection to removing this argument and moving it to its own function.

### Regarding `num_files`

I think that a `NULL` member as a sentinel is potentially more confusing and error-prone than requiring an explicit length.

I cannot think of a good way of enforcing the presence of the sentinel, and its absence would lead to very unexpected results.

Typically C APIs that accept an array expect callers to also provide a length; the big exception is strings, which have a long and sordid history of issues related to the accidental or intentional omission of `\0`.

Reply:
`eepfs_config_t` has been removed.

The option to disable the signature check has been removed to simplify the API. Upon further reflection, there is no good reason to skip the signature verification. If the signature does not match, it's likely that the filesystem will return unexpected garbage data.

Reply:
As part of integrating this PR back into the PC game port that originally spawned the idea for EEPROM FS, I have encountered an objection to removing the `const` qualifier on `eepfs_entry_t`'s `path` member: 

`ISO C++ forbids converting a string constant to 'char*' [-Wwrite-strings]`

In order to make `eepfs_config_t` compatible with C++, `path` needs to be `const char *`